src/IO/BrotliWriteBuffer.cpp

```diff
@@ -86,7 +86,7 @@ void BrotliWriteBuffer::nextImpl()
             throw Exception("brotli compress failed", ErrorCodes::BROTLI_WRITE_FAILED);
         }
     }
-    while (in_available > 0 || out_capacity == 0);
+    while (in_available > 0);
```

Comment:
Do we have similar issue with zlib and zstd?

Btw, what is the meaning of this fix?